### PR TITLE
chore(test): Add CHANGELOG + ArticleAdmin unit tests

### PR DESCRIPTION
**Pull request by srascar**
I am targetting this branch, because it's the only one

I added a CHANGELOG file within the project itself so it's easy to maintain.

- [X] Update the tests
- [ ] Update the documentation
- [ ] Add an upgrade note

## Subject
Add unit tests for Article Admin

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -38,8 +38,10 @@
         "sonata-project/block-bundle": "<3.11"
     },
     "require-dev": {
-        "matthiasnoback/symfony-config-test": "^2.0",
-        "matthiasnoback/symfony-dependency-injection-test": "^1.2",
+        "matthiasnoback/symfony-config-test": "^4.0",
+        "matthiasnoback/symfony-dependency-injection-test": "^3.0",
+        "phpunit/phpunit": "^7.2",
+        "sebastian/exporter": "3.1.0",
```
Comment (OskarStark) [inline composer.json]: Why do we need to add this explicitly?

File: tests/Admin/AdminFormFieldTestTrait.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ */
+
+namespace Sonata\ArticleBundle\Tests\Admin;
+
+use Sonata\AdminBundle\Admin\AdminInterface;
+use Sonata\AdminBundle\Form\FormMapper;
+use Symfony\Component\Form\FormBuilderInterface;
+
+/**
+ * Trait AdminFormFieldTestTrait.
```
Comment (OskarStark) [inline tests/Admin/AdminFormFieldTestTrait.php]: Please remove this doc

File: tests/Admin/AdminTestCase.php [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+use Sonata\AdminBundle\Route\RouteGeneratorInterface;
+use Sonata\AdminBundle\Security\Handler\SecurityHandlerInterface;
+use Sonata\AdminBundle\Templating\MutableTemplateRegistryInterface;
+use Sonata\AdminBundle\Translator\LabelTranslatorStrategyInterface;
+use Sonata\DoctrineORMAdminBundle\Model\ModelManager;
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+/**
+ * Class AdminTestCase.
```
Comment (OskarStark) [inline tests/Admin/AdminTestCase.php]: Please remove such docs

**Review by OskarStark — CHANGES_REQUESTED** [review]
💪🏻👍🏻

File: composer.json (hunk 1 above)
Comment (srascar) [reply composer.json]: sorry I'm going to fix

File: tests/Admin/AdminInterface.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+use Sonata\AdminBundle\Admin\AdminInterface as BaseAdminInterface;
+
+/**
+ * Add support for setBaseCodeRoute method (3.x).
+ * Add support for addParentAssociationMapping method (4.x).
+ *
+ * @author Sylvain Rascar <sylvain.rascar@ekino.com>
+ */
+interface AdminInterface extends BaseAdminInterface
```
Comment (Wharenn) [inline tests/Admin/AdminInterface.php:24]: I don't think we need this file in tests directory :)

File: src/Admin/ArticleAdmin.php [hunk 5]
```diff
@@ -91,6 +86,8 @@ public function toString($object)
     }
 
     /**
+     * @deprecated since 3.34, will be dropped in 4.0. Use TemplateRegistry services instead
```
Comment (greg0ire) [inline src/Admin/ArticleAdmin.php:89]: Version 3.34 of this bundle does not exist

File: CHANGELOG.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## Unreleased
+
+### 1.00 - 2018-08-01
+
+* Add unit tests for ArticleAdmin.
```
Comment (greg0ire) [inline CHANGELOG.md:11]: We usually use hyphens (`-`) for this

**Review by greg0ire — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -39,3 +39,6 @@ This package is available under the [MIT license](LICENSE).
 [coveralls_stable_link]: https://coveralls.io/github/sonata-project/SonataArticleBundle?branch=master
 [coveralls_unstable_badge]: https://coveralls.io/repos/github/sonata-project/SonataArticleBundle/badge.svg?branch=master
 [coveralls_unstable_link]: https://coveralls.io/github/sonata-project/SonataArticleBundle?branch=master
+
+## Note
```
Comment (OskarStark) [inline README.md:43]: Do we Need this Note?

File: src/Admin/ArticleAdmin.php (hunk 5 above)
Comment (OskarStark) [reply src/Admin/ArticleAdmin.php:89]: As this is still unreleased no need for a deprecation here

**Comment by core23** [comment]
Are you still working on this @srascar ?

**Pull request closed without merging**